### ISSUE TITLE
ERROR: Object reference not set to an instance of an object. during vod download

### ISSUE THREAD

**Issue by QcNumber1** [lay295/TwitchDownloader#841]
### Checklist

- [X] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues?q=is%3Aissue) for duplicates
- [X] I am running the latest version ([download here](https://github.com/lay295/TwitchDownloader/releases/latest))

### Edition

Windows GUI Application

### Describe your issue here

when I try to download a vod in full or just a part of it I get this Error after step 4/5 of the process

ERROR: Object reference not set to an instance of an object.

I have redowloaded to lastest version and clear cache nothing helped

I had to get back ver. 1.15.2 to make it work again

### Add any related files or extra information here

Here's the VOD ID: 1919069779

**Comment by piero0920** [comment]
I also have the same bug.

```
$ TwitchDownloaderCLI videodownload -u 1911914699 -o 'data/vod.mp4' -q best 
TwitchDownloaderCLI 1.53.3 Copyright (c) lay295 and contributors
[STATUS] - Fetching Video Info [1/5]
[LOG] - The drive '/sys' may not have enough free space to complete the download.

[STATUS] - Downloading 100% [2/5]
[STATUS] - Verifying Parts 100% [3/5]
[STATUS] - Combining Parts 100% [4/5]
[STATUS] - Finalizing Video 0% [5/5]Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at TwitchDownloaderCore.VideoDownloader.DownloadAsync(CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean )
   at System.Threading.Tasks.Task.Wait(Int32 , CancellationToken )
   at System.Threading.Tasks.Task.Wait()
   at TwitchDownloaderCLI.Modes.DownloadVideo.Download(VideoDownloadArgs inputOptions)
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
   at TwitchDownloaderCLI.Program.Main(String[] args)
Aborted (core dumped)
```

**Comment by xp906** [comment]
I just got the same error. Mine says:

8 errors were encountered while downloading: The operation was canceled. at MoveNext
ERROR: Object reference not set to an instance of an object.

Vod ID: 1901005153
(Notably this is the same vod that I could not download with version 1.53.2 because it got stuck at 100%. However the error listed above was on v1.53.3).
If anyone knows how to fix it's much appreciated!

**Comment by ScrubN** [comment]
Apologies for the problem, it has been fixed 65344c67aa95ed7d0e0fadb4d2b66a5b821b1a0a. I should really write some integration tests at some point.
